lib/register.php

```diff
@@ -332,6 +332,26 @@ function gutenberg_can_edit_post_type( $post_type ) {
 	return apply_filters( 'gutenberg_can_edit_post_type', $can_edit, $post_type );
 }
 
+/**
+ * Returns blocks from a post or content string.
+ *
+ * @since 3.7.0
+ * @see gutenberg_parse_blocks()
+ *
+ * @param int|string|WP_Post|null $post Optional. Post content, post ID, or post object. Defaults to global $post.
```

Comment:
I think this is a little too magical for me. Accepting four potential variable types that each mean different things is more confusing than it is helpful.

Given `gutenberg_parse_blocks()` already exists for parsing blocks from a string, I'd suggest `get_blocks()` should accept a `WP_Post` object, `$post_id` Post ID, or null (which would default to the global `$post`).